test/common/integreatly_stages.go

```diff
@@ -0,0 +1,124 @@
+package common
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	expectedStageProducts = map[string][]string{
+		"authentication": {
+			"rhsso",
+		},
+
+		"bootstrap": {},
+
+		"cloud-resources": {
+			"cloud-resources",
+		},
+
+		"monitoring": {
+			"middleware-monitoring",
+		},
+
+		"products": {
+			"fuse",
+			"rhssouser",
+			"datasync",
+			"codeready-workspaces",
+			"fuse-on-openshift",
+			"3scale",
+			"amqonline",
+			"ups",
+			"apicurito",
+		},
+
+		"solution-explorer": {
+			"solution-explorer",
+		},
+	}
+)
+
+func TestIntegreatlyStagesStatus(t *testing.T, ctx *TestingContext) {
+	err := wait.PollImmediate(time.Second*15, time.Minute*5, func() (bool, error) {
+		done := true
+
+		//get RHMI
+		rhmi, err := getRHMI(ctx.Client)
+		if err != nil {
+			return false, fmt.Errorf("error getting RHMI CR: %v", err)
+		}
+
+		//iterate stages and check their status
+		for stageName, productNames := range expectedStageProducts {
+			stage, ok := rhmi.Status.Stages[v1alpha1.StageName(stageName)]
+			if !ok {
+				t.Errorf("Error checking stage %s. Not found", stageName)
```

Comment:
Should it not exit out of the poll as soon as it gets an error? You never actually return an error here for this line to be called: https://github.com/integr8ly/integreatly-operator/pull/707/files#diff-0db09d831d48ec632033bf219619b029R99

What's the reason for continuing the test once it fails here?

Reply:
I don't exit the poll immediately in order to continue testing for other possible failed products or stages. However I set the `done` variable to `true`, so the poll will finish after testing all the stages

Reply:
Explained - Will continue the tests here to report all errors for all stages and products.